### DISCUSSION TITLE
HW 4 Part 3, 3a

### DISCUSSION THREAD

**Discussion by deena101** [cooperrc/computational-mechanics#62] [Q&A]
Can anyone explain what I'm doing wrong here, please and thanks!

![image](https://user-images.githubusercontent.com/106576120/176051622-c29efdc3-f70e-4ed5-a0dc-0654c258746f.png)

**Comment by cooperrc** [comment]
`z = np.block([[...]]).EUR_y` 

- [x] the `block` command creates an array 
- [ ] the `.EUR_y` is trying to run a function called `EUR_y` e.g. `EUR_y(np.block([[...]])` that's what caused the error
- [x] remove ~`.EUR_y`~

**Reply by deena101** [reply]
I'm now getting an error with the fit line. 

![image](https://user-images.githubusercontent.com/106576120/176245511-92b267be-8769-47c3-9411-51fb51475e49.png)

**Reply by cooperrc** [reply]
get rid of `.EUR_y` in all the `z` calls.

**Reply by deena101** [reply]
So would I just be left with 'fit = np.linalg.solve(z)' ? That still returns an error along with 'fit = np.linalg.solve(z@z,z)'

**Reply by cooperrc** [reply]
use

```python
fit = np.linalg.solve(z.T@z, z.T@Eur_y)
```
